multiqc/utils/report.py

```diff
@@ -363,35 +363,40 @@ def search_file(pattern, f, module_key):
     if pattern.get("contents") is not None or pattern.get("contents_re") is not None:
         if pattern.get("contents_re") is not None:
             repattern = re.compile(pattern["contents_re"])
-        try:
+        if not f.get("contents_lines") or config.filesearch_lines_limit < pattern.get("num_lines", 0):
```

Comment:
I don't really understand why we need `or config.filesearch_lines_limit < pattern.get("num_lines", 0)` to bust the cache here..?

```suggestion
        if not f.get("contents_lines"):
```

Reply:
No, wait - it's because it's for this specific search pattern. So you have to bust earlier search pattern caches, I get it..

However here we're busting the cache on every loop if the config is greater than the global setting. I think we could probably check the length of the parsed cache and check if what we have already is sufficient..

Reply:
Pushed a change here - please sanity check that it makes sense 😅